src/libraries/System.Reflection/tests/AssemblyNameTests.cs

```diff
@@ -68,6 +68,17 @@ public void Ctor_String(string name, string expectedName)
             Assert.Equal(ProcessorArchitecture.None, assemblyName.ProcessorArchitecture);
         }
 
+
+        [Theory]
+        [InlineData(" name\\u50; ", "nameP")]
+        [InlineData("name\\u50;", "nameP")]
```

Comment:
Maybe worth adding a case with null byte, like `"foo\u0000bar"` which produces `"foo"` with coreclr (and likely throws with mono).

Reply:
I was just checking what happens on mono vs. CoreClr. 
`\u` is an obscure scenario in the native assembly name parser. Since the string is wide-char anyways, the "feature" does not seem very useful and may see no real use while other implementations are unlikely to match it. Failures on mono show that. CoreRT would fail too.

We were discussing whether it may make more sense to remove this than to carry forward and to other platforms. 
Same with the strange handling of UTF16 surrogate pairs and couple other little things that CoreClr native parser tries to support but noone else supports or uses.